internal/job/vault_seal_watcher.go

```diff
@@ -29,36 +35,91 @@ func (v *VaultSealWatcher) CronSpec() string {
 }
 
 func (v *VaultSealWatcher) Run() {
-	v.log.Debug("started vault seal watcher job")
+	v.log.Debugf("started vault seal watcher job with vault HA: %v", v.conf.HAEnabled)
+
+	if v.conf.HAEnabled {
+		if len(v.conf.NodeAddresses) != 3 {
+			v.log.Errorf("vault HA node count %d is not valid", len(v.conf.NodeAddresses))
+			return
+		}
+
+		if err := v.handleUnsealForHAVault(); err != nil {
+			v.log.Errorf("%s", err)
+		}
+	} else {
+		if err := v.handleUnsealForNonHAVault(); err != nil {
+			v.log.Errorf("%s", err)
+		}
+	}
+}
+
+func (v *VaultSealWatcher) handleUnsealForNonHAVault() error {
 	vc, err := client.NewVaultClient(v.log, v.conf)
 	if err != nil {
-		v.log.Errorf("%s", err)
-		return
+		return err
 	}
 
 	res, err := vc.IsVaultSealed()
 	if err != nil {
-		v.log.Errorf("failed to get vault seal status, %s", err)
-		return
+		return fmt.Errorf("failed to get vault seal status, %s", err)
 	}
 
 	if res {
 		v.log.Info("vault is sealed, trying to unseal")
 		err := vc.Unseal()
 		if err != nil {
-			v.log.Errorf("failed to unseal vault, %s", err)
-			return
+			return fmt.Errorf("failed to unseal vault, %s", err)
 		}
 		v.log.Info("vault unsealed executed")
 
 		res, err := vc.IsVaultSealed()
 		if err != nil {
-			v.log.Errorf("failed to get vault seal status, %s", err)
-			return
+			return fmt.Errorf("failed to get vault seal status, %s", err)
 		}
 		v.log.Infof("vault sealed status: %v", res)
-		return
 	} else {
 		v.log.Debug("vault is in unsealed status")
 	}
+	return nil
+}
+
+
+func (v *VaultSealWatcher) handleUnsealForHAVault() error {
+	var vaultClients []*client.VaultClient
+	for _, nodeAddress := range v.conf.NodeAddresses {
+		conf := v.conf
+		conf.Address = nodeAddress
+		vc, err := client.NewVaultClient(v.log, conf)
+		if err != nil {
+			return err
+		}
+		vaultClients = append(vaultClients, vc)
+	}
+
+	leaderNode := v.conf.LeaderAPIAddr
+	for _, vc := range vaultClients {
+		if leader, err := vc.Leader(); err == nil && leader != "" {
+			leaderNode = leader
+			leaderCreated = true
+		}
+	}
+	v.log.Info("Found leader node: %v", leaderNode)
+
+	for index, vc := range vaultClients {
+		if leaderCreated {
+			err := vc.JoinRaftCluster(leaderNode)
+			if err != nil {
+				return fmt.Errorf("failed to join the HA cluster by node index: %v, error: %v", index, err)
+			}
+			v.log.Info("Node-%v successfully joined leader: %v", index, leaderNode)
+		}
+
+		if err := vc.Unseal(); err != nil {
```

Comment:
this also adds the leader also join the cluster, is this intended all 3 nodes got join raft clustert ?